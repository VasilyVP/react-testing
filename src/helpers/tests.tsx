//import { ReactElement } from 'react'
import { render } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom'
import TopLevelRoutes from 'routes/TopLevelRoutes'

type WithRouterProps = {
    //component: ReactElement;
    initialRoute: string;
}
export function renderRoute({ initialRoute }: WithRouterProps = { initialRoute: '/' }) {
    return render(
        <MemoryRouter initialEntries={[initialRoute]}>
            <TopLevelRoutes />
        </MemoryRouter>
    )
}
