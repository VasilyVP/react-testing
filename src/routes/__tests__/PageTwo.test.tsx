import { screen } from "@testing-library/react"
import { renderRoute } from "helpers/tests"
import userEvent from "@testing-library/user-event"


/* jest.mock('hooks/useUsers', () => ({
    __esModule: true,
    default: () => dummyUsers,
})); */

jest.mock('hooks/useUsers');

describe('PageTwo', () => {
    it('should match snapshot', () => {
        const { container } = renderRoute({ initialRoute: '/page_two' });
        expect(container).toMatchSnapshot();
    })

    it('should load the users', async () => {
        renderRoute({ initialRoute: '/page_two' });
        const liItems = await screen.findAllByRole('listitem');

        expect(liItems.length).toBe(3);
    })
    
    test('back link', () => {
        renderRoute({ initialRoute: '/page_two' });
        
        const backLink = screen.getByTestId('back_link');
        userEvent.click(backLink);
        const mp = screen.getByTestId('main_page');
        expect(mp).toBeInTheDocument();
    })
})
