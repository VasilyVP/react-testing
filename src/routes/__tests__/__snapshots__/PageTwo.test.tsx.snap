// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTwo should match snapshot 1`] = `
<div>
  <div
    data-testid="page_two"
  >
    <h1>
      Page two
    </h1>
    <h2>
      Users:
    </h2>
    <ul>
      <li>
        Leanne Graham
      </li>
      <li>
        Ervin Howell
      </li>
      <li>
        Clementine Bauch
      </li>
    </ul>
    <a
      data-testid="back_link"
      href="/"
    >
      Back
    </a>
  </div>
</div>
`;
